Code/Tools/ProjectManager/Source/PythonBindings.cpp

```diff
@@ -929,13 +929,55 @@ namespace O3DE::ProjectManager
         return AZ::Failure<AZStd::string>("Adding Gem Repo not implemented yet in o3de scripts.");
     }
 
-    GemRepoInfo PythonBindings::GemRepoInfoFromPath(pybind11::handle path, pybind11::handle pyEnginePath)
+    GemRepoInfo PythonBindings::GetGemRepoInfo(pybind11::handle repoUri)
     {
-        /* Placeholder Logic */
-        (void)path;
-        (void)pyEnginePath;
+        GemRepoInfo gemRepoInfo;
+        gemRepoInfo.m_repoLink = Py_To_String(repoUri);
 
-        return GemRepoInfo();
+        auto data = m_manifest.attr("get_repo_json_data")(repoUri);
+        if (pybind11::isinstance<pybind11::dict>(data))
+        {
+            try
+            {
+                // required
+                gemRepoInfo.m_repoLink = Py_To_String(data["repo_uri"]);
+                gemRepoInfo.m_name = Py_To_String(data["repo_name"]);
+                gemRepoInfo.m_creator = Py_To_String(data["origin"]);
+
+                // optional
+                gemRepoInfo.m_summary = Py_To_String_Optional(data, "summary", "No summary provided.");
+                gemRepoInfo.m_additionalInfo = Py_To_String_Optional(data, "additional_info", "");
+
+                auto repoPath = m_manifest.attr("get_repo_path")(repoUri);
+                gemRepoInfo.m_path = gemRepoInfo.m_directoryLink = Py_To_String(repoPath);
+
+                QString lastUpdated = Py_To_String_Optional(data, "last_updated", "");
+                gemRepoInfo.m_lastUpdated = QDateTime::fromString(lastUpdated, RepoTimeFormat);
+
+                if (data.contains("enabled"))
+                {
+                    gemRepoInfo.m_isEnabled = data["enabled"].cast<bool>();
+                }
+                else
+                {
+                    gemRepoInfo.m_isEnabled = false;
+                }
+
+                if (data.contains("gem_paths"))
+                {
+                    for (auto gemPath : data["gem_paths"])
+                    {
+                        gemRepoInfo.m_includedGemPaths.push_back(Py_To_String(gemPath));
+                    }
```

Comment:
I don't see the format of the repo.json file here.
Can you post a snippet of the one you are using?

The repo.json can contain a multiple of o3de objects including projects, templates and restricted objects: https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/repo.py#L94-L137

Reply:
We are only using repos for their gems currently but I'll get that info to you